### PR TITLE
Adds a command-line interface module

### PR DESCRIPTION
**Pull request by Baccata**
The main goal of this is to offer a way for Intellij to communicate with weaver, even if there's a mismatch of version between the IDE and the test framework 

**/!\ contains a huge json file**

### CODE REVIEW

File: modules/core/src/weaver/Log.scala [hunk 1]
```diff
@@ -54,13 +54,20 @@ object Log {
...
+  case object warn  extends Level("warn")
+  case object debug extends Level("debug")
+  case object error extends Level("error")
 
   object Level {
+    val values: List[Level] = List(info, warn, debug, error)
+
+    def fromString(s: String): Either[String, Level] =
+      values.find(_.label == s) match {
+        case other => Left(s"$other is not a valid log-level")
```
Comment (keynmol) [inline modules/core/src/weaver/Log.scala:68]: I'm confused - should there be a successful branch?

Comment (keynmol) [reply modules/core/src/weaver/Log.scala:68]:
```suggestion
        case None            => Left(s"$other is not a valid log-level")
        case Some(level) => Right(level)
```

File: modules/cli/src/weaver/cli/package.scala [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package weaver
+
+
+package object cli {
```
Comment (keynmol) [inline modules/cli/src/weaver/cli/package.scala:4]: is this needed?

File: modules/codecs/test/resources/golden.json [hunk 3]
```diff
@@ -0,0 +1,1459 @@
+{
```
Comment (Baccata) [inline modules/codecs/test/resources/golden.json:1]: This is used in a golden test, in order to prevent accidental breakage of protocol

**Pull request merged**